stream_map.go

```diff
@@ -0,0 +1,29 @@
+package astits
+
+// elementaryStreamMap represents an elementary stream ids map
+type elementaryStreamMap struct {
+	// We use map[uint32] instead map[uint16] as go runtime provide optimized hash functions for (u)int32/64 keys
+	es map[uint32]uint16 // map[StreamID]ProgramNumber
+}
+
+// newElementaryStreamMap creates a new elementary stream ids map
+func newElementaryStreamMap() *elementaryStreamMap {
+	return &elementaryStreamMap{
+		es: make(map[uint32]uint16),
+	}
+}
+
+// setLocked sets a new stream id to the elementary stream
+func (m elementaryStreamMap) setLocked(pid, number uint16) {
```

Comment:
Could you rename `Locked` to `Unlocked` in all 3 functions?

The idea is to let the developer know the function is not locking any mutex while manipulating the map which is really unsafe.